homeassistant/components/homematicip_cloud/__init__.py

```diff
@@ -52,7 +52,23 @@
     hap = HomematicipHAP(hass, entry)
     hapid = entry.data[HMIPC_HAPID].replace('-', '').upper()
     hass.data[DOMAIN][hapid] = hap
-    return await hap.async_setup()
+
+    if await hap.async_setup():
+        # Register hap as device in registry.
+        from homeassistant.helpers import device_registry as dr
```

Comment:
Put HA imports at the top of the file.

Or, alternative:

```
dev_reg = await hass.helpers.device_registry.async_get_registry()
```

Reply:
moved import to top